server/src/main/java/org/tctalent/server/service/db/impl/SavedSearchServiceImpl.java

```diff
@@ -1846,4 +1917,24 @@ private BoolQueryBuilder processElasticRequest(SearchCandidateRequest searchRequ
         return boolQueryBuilder;
     }
 
+    private String convertToJson(BoolQueryBuilder boolQueryBuilder) {
+        try {
+            XContentBuilder builder = XContentFactory.jsonBuilder();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
Analysis of this package determined that this line of code contains a resource that might not have closed properly. A resource leak can slow down or crash your system.

**Fix**
Consider closing the following resource in a try-finally block: *builder*. The resource is referenced at lines 1923, 1925. The resource is closed at line 1924. There are other execution paths that do not contain closure statements, for example, when BoolQueryBuilder.toXContent throws an exception.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).

Reply:
Good catch. I have pushed an update to use try with resources to address this potential issue.